platform/server-bundle/src/main/resources/webapp/js/services.js

```diff
@@ -2,9 +2,274 @@
     'use strict';
 
     var uiServices = angular.module('uiServices', ['ngResource']);
-
     uiServices.factory('Menu', function($resource) {
         return $resource('module/menu');
     });
 
+    uiServices.service('BootstrapDialogManager', function () {
+        var modalsList = [];
+
+        this.open = function (dialog) {
+            modalsList.push(dialog);
+            if (modalsList.length > 1) {
+                modalsList[modalsList.length-2].close();
+            }
+            dialog.open();
+        };
+
+        this.close = function (dialog) {
+            dialog.close();
+            modalsList.splice(modalsList.length-1);
+            if (modalsList.length > 0) {
+                modalsList[modalsList.length-1].open();
+            }
+        };
+
+        this.remove = function () {
+            modalsList[modalsList.length-1].close();
+            modalsList.splice(modalsList.length-1);
+            if (modalsList.length > 0) {
+                modalsList[modalsList.length-1].open();
+            }
+        };
+    });
+
+    uiServices.service('LoadingModal', function () {
+        var dialog, open = false;
+
+        this.open = function () {
+            if (!open) {
+                dialog = new BootstrapDialog({
+                    message: function(dialogRef) {
+                        var $message = $('<div></div>'),
+                        pageToLoad = dialog.getData('pageToLoad');
```

Comment:
This seems like a really long winded way to use $compile, and making a directive called <loading-splash />

**I'm ok with this being written like this way,** since we really only use the loading-splash in one place, but all the jquery DOM manipulation kinda bugs me....